src/extension/wps/wps-core/src/main/java/org/geoserver/wps/gs/GetCoveragesValue.java

```diff
@@ -0,0 +1,230 @@
+        try {
+            coordinateReferenceSystem =
+                    crs != null && !crs.isEmpty() ? CRS.decode(crs) : reader2d.getCoordinateReferenceSystem();
+        } catch (FactoryException e) {
+            throw new WPSException("Could not decode CRS " + crs, e);
+        }
+        Position position = new Position2D(coordinateReferenceSystem, x, y);
+        if (!coordinateReferenceSystem.equals(reader2d.getCoordinateReferenceSystem())) {
+            try {
+                MathTransform transform =
+                        CRS.findMathTransform(coordinateReferenceSystem, reader2d.getCoordinateReferenceSystem(), true);
+                transform.transform(position, position);
+            } catch (FactoryException | TransformException e) {
+                throw new WPSException("Unable to convert select coordinate reference system to grid CRS ", e);
+            }
+        }
+        for (String n : names) {
+            ci = catalog.getCoverageByName(n);
+            if (ci == null) {
+                throw new WPSException("Could not find coverage " + n);
+            }
+            reader = ci.getGridCoverageReader(null, null);
+            reader2d = (GridCoverage2DReader) reader;
+            if (reader2d.getOriginalEnvelope().contains(position)) {
+                try {
+                    gridCoverage2D = getGridCoverage2D(reader2d, position);
+                } catch (TransformException e) {
+                    throw new WPSException("Could not transform point " + position, e);
+                }
+                evaluationValue = convertPrimitiveArrayToNumberArray(gridCoverage2D.evaluate(position));
+                nodata = checkNoData(gridCoverage2D, evaluationValue);
+                if (!nodata) {
+                    break;
+                }
+            }
+        }
+        if (gridCoverage2D == null || nodata) {
+            return new ValuesAtPoint(new Double[] {});
+        }
+
+        return new ValuesAtPoint(evaluationValue);
+    }
+
+    private static GridCoverage2D getGridCoverage2D(GridCoverage2DReader reader, Position position)
+            throws IOException, TransformException {
+        GridCoverage2D gridCoverage2D;
+
+        Format format = reader.getFormat();
+        final ParameterValueGroup readParams = format.getReadParameters();
+        final Map<String, Serializable> parameters = CoverageUtils.getParametersKVP(readParams);
+
+        // Calculate the bbox around the point
+        MathTransform worldToGridCorner =
+                reader.getOriginalGridToWorld(PixelInCell.CELL_CORNER).inverse();
+        Coordinate coordinate = new Coordinate(position.getCoordinate()[0], position.getCoordinate()[1]);
+        JTS.transform(coordinate, coordinate, worldToGridCorner);
+        GridEnvelope2D testRange =
+                new GridEnvelope2D((int) (coordinate.getX() - 1), (int) (coordinate.getY() - 1), 2, 2);
+
+        // Calculate the world bounds of the buffered point
```

Comment:
I believe there is no need for this, use directly the GridGeometry constructor taking a pixel grid range, a grid to crs, and the target crs:

```
    public GridGeometry2D(
            final GridEnvelope gridRange, final MathTransform gridToCRS, final CoordinateReferenceSystem crs)
```

Unless you had a reason to use the other constructor, that is (in that case, please elaborate)